### DISCUSSION TITLE
Can this be used in a JVM project?

### DISCUSSION THREAD

**Discussion by rubenquadros** [orbit-mvi/orbit-mvi#79] [Q&A] [answered]
Hi @Rosomack,

Hope you are having a great day!

Can this library be used in a JVM project? I am trying it out in a JVM project but not able to get it to work. It is always requiring 
```org.orbitmvi.orbit.viewmodel.container``` .

I have currently only imported ```implementation("org.orbit-mvi:orbit-core:4.2.0")``` dependency which should be enough?
Also, have tried following the example here: https://orbit-mvi.org/Core/overview

Please suggest. Thanks in advance!

**Comment by Rosomack** [comment, marked as answer]
@rubenquadros I think you've found a bug in our documentation!

The example is wrong - the `container` factory function is an extension on `CoroutineScope` so you need a scope to work with it.

I'll correct the docs, here's a proper example:

``` kotlin
data class ExampleState(val seen: List<String> = emptyList())

sealed class ExampleSideEffect {
   data class Toast(val text: String)
}

class ExampleContainerHost(scope: CoroutineScope): ContainerHost<ExampleState, ExampleSideEffect> {
    
    // create a container
    override val container = scope.container<ExampleState, ExampleSideEffect>(ExampleState())

    fun doSomethingUseful() = intent {
        ...
    }
}

private val scope = CoroutineScope(Dispatchers.Main)
private val viewModel = ExampleContainerHost(scope)

fun main() {

    // subscribe to updates
    // On Android, use ContainerHost.observe() from the orbit-viewmodel module
    scope.launch {
        viewModel.stateFlow.collect {
            // do something with the state
        }
    }
    scope.launch {
        viewModel.sideEffectFlow.collect {
            // do something with the side effect
        }
    }

    viewModel.doSomethingUseful()
    
    // Ensure the main function does not complete so we can do something useful with the container.
}


```

That said this is a very synthetic example, the main function might not be the best place to create containers within a pure JVM app.
